src/Effect/Ref.purs

```diff
@@ -20,7 +20,40 @@ import Effect (Effect)
 -- | which holds a value of type `a`.
 foreign import data Ref :: Type -> Type
 
+type role Ref representational
+
 -- | Create a new mutable reference containing the specified value.
+-- |
+-- | ## Why This Type Signature Cannot Be `forall s. s -> Ref s`
```

Comment:
I think a separate subheading here is overkill, especially since there's only one line of text before it.